pkg/agent/rest.go

```diff
@@ -34,27 +36,38 @@ func NewNodeZeroRestClient(ctx context.Context, assetDir string) (*NodeZeroRestC
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create asset store")
 	}
-	nmState := &manifests.NMStateConfig{}
-	if err := assetStore.Fetch(nmState); err != nil {
-		return nil, errors.Wrapf(err, "failed to fetch %s", nmState.Name())
+
+	agentConfig, agentConfigError := assetStore.Load(&agentconfig.AgentConfig{})
+	agentManifests, manifestError := assetStore.Load(&manifests.AgentManifests{})
+
+	if agentConfigError != nil {
+		logrus.Debug(errors.Wrapf(agentConfigError, "failed to load %s", agentConfig.(*agentconfig.AgentConfig).Name()))
+	}
+	if manifestError != nil {
+		logrus.Debug(errors.Wrapf(manifestError, "failed to load %s", agentManifests.(*manifests.AgentManifests).Name()))
 	}
 
-	NodeZeroIP, err := manifests.GetNodeZeroIP(nmState.Config)
+	if agentConfigError != nil && manifestError != nil {
+		return nil, errors.New("failed to load AgentConfig and NMStateConfig")
+	}
+
+	RendezvousIP, err := image.RetrieveRendezvousIP(agentConfig.(*agentconfig.AgentConfig).Config, agentManifests.(*manifests.AgentManifests).NMStateConfigs)
```

Comment:
Just as a reminder note (not for this PR), I think in future it could be useful (at least for the maintenance) to deploy a (non-writable) RendezvousIP asset (with a dependency on AgentConfig and AgentManifests - if not directly NMStateConfig), to be used directly here and in the Ignition asset

Reply:
I like this idea. After going through this exercise I think we can consolidate the logic for the RendezvousIP into it's own asset that will clean it up a lot for the future.